language-design.md

```diff
@@ -26,32 +30,32 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
----
+This is a bit of a tough question because what makes a language well or poorly designed sometimes depends on the metrics that we use. One can argue R is a well-designed language, but Tim Smith found it difficult to learn, and so disliked it even though he now regrets the tone he took in intitial criticisms. [Smith, 2016] Perl is very difficult to learn as well, and yet is still relevant today as a programming language. [Stefik _et al._, 2011] Therefore, metrics like ease-of-use for a novice programmer can be subjective and are one symptom of a language. That being said, I think there are some things I use when I try to figure out whether a language is well-designed or not. I definitely will judge it based on whether the rules feel consistent. Python, C++, and R all have different means and rules for assignments, but once you know the rules you can always assign something. Similarly, if it's consistent than I also feel like it's a little predictable - little syntax things I might have to look up, but I can follow what's happening in a program. A poorly designed language, I cannot.
```

Comment:
Humans are good at recognizing patterns, but that requires for a reasonable pattern to exist in the first place. For example, take regex; it is definitely all about pattern matching and has consistent rules, but it's a nightmare to decipher. For that reason, I've seen some papers floating around about introducing more natural language to regex, but here we are in 2023, still using regex as the standard for pattern-matching...I guess what this tangent is trying to say is that consistency is good in conjunction with fluency, something that you touch on briefly at the end by mentioning "I can follow what's happening in a program."